client/commands/query/query_test.go

```diff
@@ -0,0 +1,131 @@
+package query
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/go-wire"
+	lc "github.com/tendermint/light-client"
+	"github.com/tendermint/light-client/certifiers"
+	certclient "github.com/tendermint/light-client/certifiers/client"
+	nm "github.com/tendermint/tendermint/node"
+	"github.com/tendermint/tendermint/rpc/client"
+	rpctest "github.com/tendermint/tendermint/rpc/test"
+	"github.com/tendermint/tendermint/types"
+	"github.com/tendermint/tmlibs/log"
+
+	"github.com/tendermint/basecoin/app"
+	"github.com/tendermint/basecoin/modules/eyes"
+)
+
+var node *nm.Node
+
+func TestMain(m *testing.M) {
+	logger := log.TestingLogger()
+	store, err := app.NewStore("", 0, logger)
+	if err != nil {
+		panic(err)
+	}
+	app := app.NewBasecoin(eyes.NewHandler(), store, logger)
+	node = rpctest.StartTendermint(app)
+
+	code := m.Run()
+
+	node.Stop()
+	node.Wait()
+	os.Exit(code)
+}
+
+func TestAppProofs(t *testing.T) {
+	assert, require := assert.New(t), require.New(t)
+
+	cl := client.NewLocal(node)
+	client.WaitForHeight(cl, 1, nil)
+
+	k := []byte("my-key")
+	v := []byte("my-value")
+
+	tx := eyes.SetTx{Key: k, Value: v}.Wrap()
+	btx := wire.BinaryBytes(tx)
+	br, err := cl.BroadcastTxCommit(btx)
+	require.Nil(err, "%+v", err)
+	require.EqualValues(0, br.CheckTx.Code, "%#v", br.CheckTx)
+	require.EqualValues(0, br.DeliverTx.Code)
+
+	// This sets up our trust on the node based on some past point.
+	source := certclient.New(cl)
+	seed, err := source.GetByHeight(br.Height - 2)
+	require.Nil(err, "%+v", err)
+	cert := certifiers.NewStatic("my-chain", seed.Validators)
+
+	latest, err := source.GetLatestCommit()
+	require.Nil(err, "%+v", err)
+	rootHash := latest.Header.AppHash
+
+	// Test existing key.
+	var data eyes.Data
+
+	bs, height, proofExists, _, err := getWithProof(k, cl, cert)
+	require.Nil(err, "%+v", err)
+	require.NotNil(proofExists)
+	require.True(height >= uint64(latest.Header.Height))
+
+	err = wire.ReadBinaryBytes(bs, &data)
+	require.Nil(err, "%+v", err)
+	assert.EqualValues(v, data.Value)
+	err = proofExists.Verify(k, bs, rootHash)
+	assert.Nil(err, "%+v", err)
+
+	// Test non-existing key.
+	missing := []byte("my-missing-key")
+	bs, _, proofExists, proofNotExists, err := getWithProof(missing, cl, cert)
+	require.True(lc.IsNoDataErr(err))
+	require.Nil(bs)
+	require.Nil(proofExists)
+	require.NotNil(proofNotExists)
+	err = proofNotExists.Verify(missing, rootHash)
+	assert.Nil(err, "%+v", err)
```

Comment:
you can use `assert.NoError` here https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1012